### PR TITLE
(maint) Using && and || can cause issues

### PR DESCRIPTION
**Pull request by underscorgan**
Care needs to be taken when combining && and || in a long command to
avoid running into operator precedence issues.

### CODE REVIEW

File: lib/packaging/deb/repo.rb [hunk 1]
```diff
@@ -83,7 +83,7 @@ def retrieve_repo_configs(target = "repo_configs")
     end
 
     def repo_creation_command(repo_directory, artifact_paths)
-      cmd = "[ -d #{repo_directory} ] || exit 1 && "
+      cmd = "[ -d #{repo_directory} ] || exit 1 ; "
```
Comment (sarameisburger) [inline lib/packaging/deb/repo.rb]: What was the purpose of having the && in the first place?

Comment (underscorgan) [reply lib/packaging/deb/repo.rb]: Looking back in the original commit all the commands used to end in `;` and were switched to `&&` so a command fails in the chain we don't end up running a command in an unplanned location or end up in a weird state if we failed a prerequisite command or something.

Comment (underscorgan) [reply lib/packaging/deb/repo.rb]: (https://github.com/puppetlabs/packaging/pull/693/commits/b17bc443a5c48454a693da9ddbd2988466405792 for reference)

Comment (sarameisburger) [reply lib/packaging/deb/repo.rb]: Ah, interesting. Would using 'and' be useful at all? Stack Overflow tells me it has lower precedence

Comment (underscorgan) [reply lib/packaging/deb/repo.rb]: I don't think bash has `and` (these are going to be executed in bash via an ssh command)

Comment (sarameisburger) [reply lib/packaging/deb/repo.rb]: Ohh good to know. Thanks for all the info!

**Pull request merged**